src/reanimated2/frameCallback/FrameCallbackRegistry.ts

```diff
@@ -1,54 +1,107 @@
-export default class FrameCallbackRegistry {
-  private nextCallbackId = 0;
-  private frameCallbackRegistry = new Map<number, () => void>();
-  private frameCallbackActive = new Set<number>();
-  private isFrameCallbackRunning = false;
-
-  private runAnimation() {
-    const loop = () => {
-      this.frameCallbackActive.forEach((key) => {
-        const callback = this.frameCallbackRegistry.get(key);
-        callback!();
-      });
-
-      if (this.frameCallbackActive.size > 0) {
+import { runOnUI } from '../';
+
+export interface FrameCallbackRegistryUI {
+  frameCallbackRegistry: Map<number, () => void>;
+  frameCallbackActive: Set<number>;
+  isFrameCallbackRunning: boolean;
+  runCallbacks: () => void;
+  registerFrameCallback: (callback: () => void, callbackId: number) => void;
+  unregisterFrameCallback: (frameCallbackId: number) => void;
+  manageStateFrameCallback: (frameCallbackId: number, state: boolean) => void;
+}
+
+const prepareUIRegistry = runOnUI(() => {
+  'worklet';
+
+  const frameCallbackRegistry: FrameCallbackRegistryUI = {
+    frameCallbackRegistry: new Map<number, () => void>(),
+    frameCallbackActive: new Set<number>(),
+    isFrameCallbackRunning: false,
+
+    runCallbacks() {
+      const loop = () => {
+        this.frameCallbackActive.forEach((key: number) => {
+          const callback = this.frameCallbackRegistry.get(key);
+          callback!();
+        });
+
+        if (this.frameCallbackActive.size > 0) {
+          requestAnimationFrame(loop);
+        } else {
+          this.isFrameCallbackRunning = false;
+        }
+      };
+
+      if (!this.isFrameCallbackRunning) {
         requestAnimationFrame(loop);
+        this.isFrameCallbackRunning = true;
+      }
+    },
+
+    registerFrameCallback(callback: () => void, callbackId: number) {
+      this.frameCallbackRegistry.set(callbackId, callback);
+    },
+
+    unregisterFrameCallback(frameCallbackId: number) {
+      this.manageStateFrameCallback(frameCallbackId, false);
+      this.frameCallbackRegistry.delete(frameCallbackId);
+    },
+
+    manageStateFrameCallback(frameCallbackId: number, state: boolean) {
+      if (frameCallbackId === -1) {
+        return;
+      }
+      if (state) {
+        this.frameCallbackActive.add(frameCallbackId);
+        this.runCallbacks();
       } else {
-        this.isFrameCallbackRunning = false;
+        this.frameCallbackActive.delete(frameCallbackId);
       }
-    };
+    },
+  };
+
+  global._frameCallbackRegistry = frameCallbackRegistry;
+});
+
+export default class FrameCallbackRegistryJS {
+  private nextCallbackId = 0;
 
-    if (!this.isFrameCallbackRunning) {
-      requestAnimationFrame(loop);
-      this.isFrameCallbackRunning = true;
+  constructor() {
+    if (global._frameCallbackRegistry === undefined) {
```

Comment:
This code base is common for all platforms. I would rather avoid injecting unnecessary data into global objects on the Web platform.